ws-client/src/client.rs

```diff
@@ -320,8 +322,13 @@ async fn background_task(
 				Some(Ok(jsonrpc::Response::Single(response))) => {
 					match process_response(&mut manager, response, max_capacity_per_subscription) {
 						Ok(Some((unsubscribe, params))) => {
-							if let Err(e) = sender.start_request(unsubscribe, params).await {
-								log::error!("Failed to send unsubscription response: {:?}", e);
+							match sender.start_request(unsubscribe, params).await {
+								Ok(id) => {
+									if let Err(e) = manager.insert_pending_call(id, None) {
```

Comment:
Ideally, move the request manager to the sender instead to avoid having to this manual inserting.

In addition, if the request ID is already occupied leaves the client in a weird state i.e, might map the request to a request id. Also an argument why to move the request manager to the `JSONRPC Sender`, i.e, might be better to check if the request ID is not in the request manager and to avoid sending another request with the same ID. However, that's very unlikely one would need to send more requests than u64::MAX for that to happen.